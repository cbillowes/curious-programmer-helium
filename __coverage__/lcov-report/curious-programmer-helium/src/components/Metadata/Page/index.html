<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for curious-programmer-helium/src/components/Metadata/Page</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> curious-programmer-helium/src/components/Metadata/Page
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">82.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>33/40</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.29% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>18/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">82.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>33/40</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="Metadata.jsx"><a href="Metadata.jsx.html">Metadata.jsx</a></td>
	<td data-value="82.5" class="pic high"><div class="chart"><div class="cover-fill" style="width: 82%;"></div><div class="cover-empty" style="width:18%;"></div></div></td>
	<td data-value="82.5" class="pct high">82.5%</td>
	<td data-value="40" class="abs high">33/40</td>
	<td data-value="64.29" class="pct medium">64.29%</td>
	<td data-value="28" class="abs medium">18/28</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	<td data-value="82.5" class="pct high">82.5%</td>
	<td data-value="40" class="abs high">33/40</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jan 05 2019 22:01:48 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
