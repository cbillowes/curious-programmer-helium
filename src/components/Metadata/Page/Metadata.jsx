import React, { Component } from "react"
import Helmet from "react-helmet"
import { graphql } from "gatsby";
import config from "../../../../data/SiteConfig"
import navigation from "../../../../data/Navigation"
import Path from "path"
import url from "url-join"

class Metadata extends Component {
  constructor(props) {
    super(props)
    this.state = {
      slug: (this.props.slug) ? this.props.slug : this.getSlug(props)
    }
  }

  getSlug = (props) => {
    if (props.data.slug) {
      return edges[0].node.fields.slug
    }
    return "/"
  }

  getTitle = (title) => {
    if (title && title !== config.siteTitle) {
      return `${title} | ${config.siteTitle}`
    }

    if (!title) {
      const navigationItem = getNavigationItem(this.state.slug)
      return `${navigationItem.name} | ${config.siteTitle}`
    }

    return config.siteTitle
  }

  getDescription = (description) => {
    if (description) {
      return description
    }
    return config.siteDescription
  }

  getImageUrl = (images, slug) => {
    if (images) {
      let image
      if (images.ogImage) {
        image = images.ogImage
      } else if (images.cover) {
        image = images.cover
      } else if (config.defaultOgImage) {
        image = config.defaultOgImage
      } else {
        image = config.siteLogo
      }

      if (image.startsWith("http")) {
        return image
      }
      if (image.startsWith("/")) {
        return config.siteUrl + Path.resolve(slug, image)
      }
      return config.siteUrl + Path.resolve(slug, image)
    }
    return config.siteUrl + Path.resolve(slug, config.siteLogo)
  }

  getUrl = (slug) => {
    if (slug) {
      return url(config.siteUrl, slug)
    }
    let navigationItem = getNavigationItem(this.state.slug)
    let pageUrl = url(config.siteUrl, navigationItem.url)
    return pageUrl
  }

  getType = (slug) => {
    if (slug && slug.indexOf("/blog/")) {
      return "article"
    }
    return "website"
  }

  render() {
    const { title, description, images, slug } = this.props
    const pageTitle = this.getTitle(title)
    const pageDescription = this.getDescription(description)
    const imageUrl = this.getImageUrl(images, slug)
    const pageUrl = this.getUrl(slug)
    const pageType = this.getType(slug)

    return (
      <Helmet>
        <title>{pageTitle}</title>
        <link rel="canonical" href={pageUrl} />
        <link rel="icon" href="/favicon.png" />
        <meta name="description" content={pageDescription} />
        <meta name="image" content={imageUrl} />

        <meta name="twitter:card" content="summary_large_image" />
        <meta name="twitter:site" content={config.userTwitter} />
        <meta name="twitter:creator" content={config.userTwitter} />

        <meta property="fb:app_id" content={config.siteFBAppID} />
        <meta property="og:title" content={pageTitle} />
        <meta property="og:description" content={pageDescription} />
        <meta property="og:image" content={imageUrl} />
        <meta property="og:url" content={pageUrl} />
        <meta property="og:type" content={pageType} />

        <script type="application/ld+json">
          {getSchema(pageTitle, pageDescription, pageUrl, imageUrl)}
        </script>
      </Helmet>
    );
  }
}

function getNavigationItem(slug) {
  let navigationItem = navigation[0]
  navigation.map((item) => {
    if (slug.indexOf(item.url) > -1) {
      navigationItem = item
    }
    return navigationItem
  })
  return navigationItem
}

function getSchema(title, description, url, image) {
  const schema = [
    {
      "@context": "http://schema.org",
      "@type": "WebSite",
      url: url,
      name: config.siteTitle,
      alternateName: config.siteTitleAlt
    },
    {
      "@context": "http://schema.org",
      "@type": "BreadcrumbList",
      itemListElement: [
        {
          "@type": "ListItem",
          position: 1,
          item: {
            "@id": url,
            name: title,
            image: image
          }
        }
      ]
    },
    {
      "@context": "http://schema.org",
      "@type": "BlogPosting",
      url: url,
      name: title,
      alternateName: config.siteTitleAlt,
      headline: title,
      image: {
        "@type": "ImageObject",
        url: image
      },
      description: description
    }
  ]
  return JSON.stringify(schema)
}

export default Metadata

export const pageQuery = graphql`
  query MetadataComponent($slug: String!) {
    allMarkdownRemark {
      edges {
        node {
          fields {
            slug
          }
        }
      }
    }
    markdownRemark(frontmatter: { slug: { eq: $slug } }) {
      fields {
        slug
      }
    }
  }
`