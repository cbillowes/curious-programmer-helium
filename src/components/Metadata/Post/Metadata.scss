@import "../../../layout/layout.scss";

.post-metadata {
  p {
    font-style: italic;
    text-align: right;
    font-size: 0.8em;
  }

  .tags {
    text-align: center;
  }
}