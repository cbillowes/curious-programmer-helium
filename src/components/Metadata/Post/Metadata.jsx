import React, { Component } from "react"
import Moment from "react-moment"
import Tags from "../../Tag/Tags"
import "./Metadata.scss"

class Metadata extends Component {
  render() {
    const { tags, date, timeToRead} = this.props

    return (
      <div className="post-metadata">
        <p>
          posted { " " }
          <Moment date={date} fromNow /> { " " }
          on { " " }
          <Moment date={date} format="ddd, D MMMM YYYY" /> { " " }
          (Est. {timeToRead} minute read)
        </p>
        {(tags) ? <Tags tags={tags} /> : <span> </span>}
      </div>
    )
  }
}

export default Metadata