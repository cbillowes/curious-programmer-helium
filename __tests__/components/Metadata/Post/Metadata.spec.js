import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Metadata from "../../../../src/components/Metadata/Post/Metadata"

describe("a post metadata component", () => {
  it("should render", () => {
    const component = <Metadata />
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render the attributes of the post written", () => {
    let date = new Date(2010, 0, 1)
    let component = mount(<Metadata date={date} />)
    expect(component.html()).toContain(`Fri, 1 January 2010`)

    date = new Date(2010, 0, 21)
    component = mount(<Metadata date={date} />)
    expect(component.html()).toContain(`Thu, 21 January 2010`)

    date = new Date()
    component = mount(<Metadata date={date} />) 
    expect(component.html()).toContain(`a few seconds ago</time>`)

    component = mount(<Metadata timeToRead={10} />) 
    expect(component.html()).toContain(`(Est. 10 minute read)`)

    component = mount(<Metadata tags={["technical"]} />) 
    expect(component.html()).toContain(`<ul class="tags">`)
  })
})