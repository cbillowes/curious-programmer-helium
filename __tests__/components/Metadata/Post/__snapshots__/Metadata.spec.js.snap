// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a post metadata component should render 1`] = `
<div
  className="post-metadata"
>
  <p>
    posted 
     
    <time
      dateTime={"2019-01-05T06:58:04.680Z"}
    >
      a few seconds ago
    </time>
     
     
    on 
     
    <time
      dateTime={"2019-01-05T06:58:04.680Z"}
    >
      Sat, 5 January 2019
    </time>
     
     
    (Est. 
     minute read)
  </p>
  <span>
     
  </span>
</div>
`;
